httpx/_decoders.py

```diff
@@ -44,7 +50,8 @@ class DeflateDecoder(Decoder):
     See: https://stackoverflow.com/questions/1838699
     """
 
-    def __init__(self) -> None:
+    def __init__(self, request: "Request") -> None:
+        self.request = request
```

Comment:
? (Applicable to all other decoders here)

```suggestion
        super().__init__(request)
```

Reply:
Yeah a guess a lot of other Python developers would *tend* always just rely on "hey I'm supposed to call `super` right". Personally, I think explicit is preferable to indirect here. Calling into `super` isn't always necessary or inherently the "right" thing to do.

Reply:
I see, well then I guess `Decoder` would really be better served by a `Protocol`, rather than a full-fledged base class… But we only have them on Py38+, which is unfortunate.

```python
class Decoder(Protocol):
    request: Request

    def decode(self, data: bytes) -> bytes:
        ...

    def flush(self) -> bytes:
        ...
```

So maybe we could then also use the `request: Request` annotation on the `Decoder` interface, and _also_ add an explicit constructor for `IdentityDecoder`?

Reply:
`MultiDecoder` inherits the `Decoder` interface, but do not have this property